evap/staff/tests/test_views.py

```diff
@@ -2280,6 +2280,48 @@ def test_exclude_unfinished_evaluations(self):
             # unfinished because still in EVALUATION_END_OFFSET_HOURS
             self.assertNotContains(page, self.evaluation2.full_name)
 
+    def test_suggested_evaluation_ordering(self):
+        evaluations = baker.make(
+            Evaluation,
+            course__semester=self.semester,
+            participants=[self.student1, self.student2],
+            voters=[self.student1, self.student2],
+            state=Evaluation.State.IN_EVALUATION,
+            vote_start_datetime=datetime.datetime.now() - datetime.timedelta(days=42),
+            vote_end_date=datetime.date.today() - datetime.timedelta(days=2),
+            can_publish_text_results=True,
+            _quantity=2,
+        )
+
+        for evaluation, answer_count in zip(evaluations, [1, 2]):
+            contribution = baker.make(Contribution, evaluation=evaluation, _fill_optional=["contributor"])
+            baker.make(TextAnswer, contribution=contribution, question__type=Question.TEXT, _quantity=answer_count)
+
+        url = f"/staff/semester/{self.semester.pk}/evaluation/{self.evaluation2.pk}/textanswers"
+
+        with run_in_staff_mode(self):
+            # Since Evaluation 1 has an extra text answer, it should be first
+            page = self.app.get(url, user=self.manager)
+            self.assertTrue(
+                f'data-evaluation="{evaluations[1].pk}"' in str(page.html.select("span[data-next-evaluation-index]")[0])
+            )
```

Comment:
my last suggestion: You can use `assertIn` and `select_one`

```patch
diff --git a/evap/staff/tests/test_views.py b/evap/staff/tests/test_views.py
index ada2c353..9f7b9019 100644
--- a/evap/staff/tests/test_views.py
+++ b/evap/staff/tests/test_views.py
@@ -2302,24 +2302,27 @@ class TestEvaluationTextAnswerView(WebTest):
         with run_in_staff_mode(self):
             # Since Evaluation 1 has an extra text answer, it should be first
             page = self.app.get(url, user=self.manager)
-            self.assertTrue(
-                f'data-evaluation="{evaluations[1].pk}"' in str(page.html.select("span[data-next-evaluation-index]")[0])
+            self.assertIn(
+                f'data-evaluation="{evaluations[1].pk}"',
+                str(page.html.select_one("span[data-next-evaluation-index]")),
             )
 
             # Since Evaluation 0 has an earlier end date, it should now be first
             evaluations[0].vote_end_date = datetime.date.today() - datetime.timedelta(days=4)
             evaluations[0].save()
             page = self.app.get(url, user=self.manager)
-            self.assertTrue(
-                f'data-evaluation="{evaluations[0].pk}"' in str(page.html.select("span[data-next-evaluation-index]")[0])
+            self.assertIn(
+                f'data-evaluation="{evaluations[0].pk}"',
+                str(page.html.select_one("span[data-next-evaluation-index]")),
             )
 
             # Since the grading process for Evaluation 1 is finished, it should be first
             evaluations[1].wait_for_grade_upload_before_publishing = False
             evaluations[1].save()
             page = self.app.get(url, user=self.manager)
-            self.assertTrue(
-                f'data-evaluation="{evaluations[1].pk}"' in str(page.html.select("span[data-next-evaluation-index]")[0])
+            self.assertIn(
+                f'data-evaluation="{evaluations[1].pk}"',
+                str(page.html.select_one("span[data-next-evaluation-index]")),
             )
 
     def test_num_queries_is_constant(self):
```

Reply:
(You can apply this patch by saving the entire thing (starting with the `diff --git`) in a file (say, `x.patch`) and then running `git apply x.patch` and don't need to retype everything if you don't want to :) )